docs/src/main/sphinx/admin/properties.rst

```diff
@@ -21,3 +21,80 @@ may be used to tune Trino or alter its behavior when required.
     Logging <properties-logging>
     Web UI <properties-web-interface>
     Regular expression function <properties-regexp-function>
+
+Property value types
+--------------------
+
+Configuration properties in Trino take values of different types, some standard
+and some non-standard. Supported values may be further limited on a per-property
+basis.
+
+.. _prop-type-boolean:
+
+``boolean``
+^^^^^^^^^^^
+
+Properties of a ``boolean`` type support two values, ``true`` or ``false``
+in lowercase.
+
+.. _prop-type-data-size:
+
+``data size``
+^^^^^^^^^^^^^
+
+Properties of a ``data size`` type support values describing an amount of
+data, measured in byte-based units. For example, the value ``6GB`` describes
+six gigabytes.
+
+The ``data size`` type supports the following units:
+
+* ``B``: Bytes
+* ``kB``: Kilobytes
+* ``MB``: Megabytes
+* ``GB``: Gigabytes
+* ``TB``: Terabytes
+* ``PB``: Petabytes
+
+Some ``data size`` properties also support ``unlimited`` as a value.
+
+.. _prop-type-double:
+
+``double``
+^^^^^^^^^^
+
+Properties of a ``double`` type support numerical values, including decimals.
+
+.. _prop-type-duration:
+
+``duration``
+^^^^^^^^^^^^
+
+Properties of a ``duration`` type support values describing an amount of time.
+For example, the value ``7m`` describes seven minutes.
+
+The ``duration`` type supports the following units:
```

Comment:
needs examples .. also explain if you can do something like 2h30m 

also what about months and years?

Reply:
Months and years are not defined in the Airlift class. Unsure if we support composite durations like that, @hashhar @findepi ?

Reply:
Also,  how do we take values of 0 duration? Just ``0``, ``0s``, etc

Reply:
Duration only has the units - nanos (ns), micros (us), millis (ms), seconds (s), minutes (m), hours (h) and days (d).

zero-durations can be any valid duration with the value 0. So each of these mean the same thing `0ms`, `0s`, `0m`, `0d` etc.

Reply:
Please mention a duration must have a single unit - ie. no composition. Looks good otherwise.

Reply:
> duration types support the syntax of a number followed by a unit

Adding this would alleviate my concerns. Composition is a natural assumption/question.
Something like:

```
The properties of type ``duration`` support values describing an
amount of time and take the form of a number followed by a unit.
```

Reply:
Generally agreed about not documenting everything we DON'T support - that's kinda impossible and too verbose.
In this case I think it's a valid assumption that a lot of people would make.

Reply:
Sounds good - made the appropriate syntax more explicit.